modules/gdscript/gdscript_parser.cpp

```diff
@@ -3644,16 +3635,55 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 	variable->export_info.type = t_type;
 	variable->export_info.hint = t_hint;
 
+	const int64_t max_flags = 32;
+
+	if (p_annotation->name == SNAME("@export_flags") && p_annotation->resolved_arguments.size() > max_flags) {
+		push_error(vformat(R"(The argument count limit for "@export_flags" is exceeded (%d/%d).)", p_annotation->resolved_arguments.size(), max_flags), p_annotation);
+		return false;
+	}
```

Comment:
I think this code is no longer needed? Since the flag values are already checked below, and (in theory) there can be more than 32 arguments (different combinations of flags)?

Reply:
The value of each argument can't exceed 32^2 because the underlying type is a 32-bit integer. So this error is still useful because it helps avoid exceeding this limit when the arguments don't specify their value, although it could be improved.

Reply:
No, I mean something like:

```
A:1,B:2,C:4,D:8,E:16,F:32,G:64,<X>

```

Where `<X>` is all combinations of 3 flags. `C(7,3) = 7! / (3! * 4!) = 35`. So there are `7 + 35 = 42` arguments in total. This is greater than 32, but each of the arguments does not exceed the allowed value.

The only thing missing from the check below is the calculation of an auto value if `:number` is not specified.

Note: I haven't checked how the inspector handles this.

Reply:
My response was addressing your question `I think this code is no longer needed?`. I was thinking it was still needed for the case when values are not specified because it will increase the values automatically so the 33rd argument will exceed 2^32 (4294967296). But I totally missed that you also said `Since the flag values are already checked below`.

I think, when the value is not specified, this check would give a better error. Otherwise, it may be a bit confusing for users to get `Invalid argument %d of annotation "@export_flags": The flag value must be at least 1 and at most 2 ** %d - 1.` when they didn't specify any value.

Reply:
I think we can finalize this PR, and discuss further improvements separately.

Reply:
```gdscript
@export_flags(\
"F00:1", "F01:1", "F02:1", "F03:1",
"F04:1", "F05:1", "F06:1", "F07:1",
"F08:1", "F09:1", "F10:1", "F11:1",
"F12:1", "F13:1", "F14:1", "F15:1",
"F16:1", "F17:1", "F18:1", "F19:1",
"F20:1", "F21:1", "F22:1", "F23:1",
"F24:1", "F25:1", "F26:1", "F27:1",
"F28:1", "F29:1", "F30:1", "F31:1",
"F32:2", "F33:4", "F34:8", "F35:16"
) var test
```